src/components/Table/hoc/withTableSettings/TableColumnSetup/TableColumnSetup.tsx

```diff
@@ -365,7 +378,7 @@ export const TableColumnSetup = (props: TableColumnSetupProps) => {
     return (
         <TreeSelect
             className={b(null, className)}
-            mapItemDataToProps={identity}
+            mapItemDataToProps={mapItemDataToProps}
```

Comment:
Why is `dndItems` needed? `mapItemDataToProps` translates items to props, no need to prepare items for that.